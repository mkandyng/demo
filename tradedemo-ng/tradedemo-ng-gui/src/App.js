import React from "react";
import { Route, NavLink, HashRouter } from "react-router-dom";
import TradingDemo from "./routes/tradingDemo";
import Documentation from "./routes/documentation";

export default function Apps() {
    return (
       <HashRouter>
         <div>
           <ul className="header">
             <li><NavLink exact to="/">Trading Demo</NavLink></li>
             <li><NavLink to="/documentation">Documentation</NavLink></li>
           </ul>
           <div className="content">
               <Route exact path="/" component={TradingDemo}/>
               <Route path="/Documentation" component={Documentation}/>
           </div>
         </div>
      </HashRouter>
    )
}
