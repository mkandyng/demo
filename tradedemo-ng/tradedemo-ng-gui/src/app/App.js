import React from "react";
import { Route, NavLink, HashRouter } from "react-router-dom";
import TradingDemo from "../pages/TradingDemo/TradingDemo";
import Documentation from "../pages/Documentation/Documentation";

/**
 * [Apps Application Main defining routes, separating between header and content]
 */
export default function Apps() {
    return (
       <HashRouter>
           <div>
               <ul className="header">
                   <li><NavLink exact to="/">Trading Demo</NavLink></li>
                   <li><NavLink to="/documentation">Documentation</NavLink></li>
               </ul>
               <div className="content">
                   <Route exact path="/" component={TradingDemo}/>
                   <Route path="/Documentation" component={Documentation}/>
               </div>
           </div>
      </HashRouter>
    )
}
