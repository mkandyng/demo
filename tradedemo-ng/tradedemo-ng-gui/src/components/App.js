import React from "react";
import { Route, NavLink, HashRouter } from "react-router-dom";
import TradingViewContainer from "./tradingView/TradingViewContainer";
import DocumentationContainer from "./documentation/DocumentationContainer";

const App = function(props) {
    return (
	<HashRouter>
            <div>
                <ul className="header">
                    <li><NavLink exact to="/">Trading View</NavLink></li>
                    <li><NavLink to="/documentation">Documentation</NavLink></li>
                </ul>
                <div className="content">
		    <Route exact path="/" component={TradingViewContainer}/>
		    <Route path="/Documentation" component={DocumentationContainer}/>
                </div>
            </div>
	</HashRouter>
    )
};

export default App;
