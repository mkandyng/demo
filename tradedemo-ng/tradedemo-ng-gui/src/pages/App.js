import React from "react";
import {Provider} from "react-redux";
import {Route, NavLink, HashRouter} from "react-router-dom";
import {getRandomInt} from "../libs/utils";
import {fetchInstruments, updateMarketfeedInstrument} from "../modules/instruments/store/instrumentsActions";
import {flashPriceUpdate} from "../modules/instruments/instruments";
import TradingDemo from "./TradingDemo";
import Documentation from "./Documentation";

export default function appWithProvider(store) {
  initOnStartup(store);
  return (<Provider store={store}>
    <App/>
  </Provider>)
};

function initOnStartup(store) {
  store.dispatch(fetchInstruments());
    flashPriceOnMarketFeed(store);
    PlaceSampleOrders(store, 10);
}

function flashPriceOnMarketFeed(store) {
  setInterval(() => {
    if (store.getState().instruments &&
        store.getState().instruments.selected.symbol !== undefined) {
      flashPriceUpdate(store.getState().instruments.marketfeedInstruments,
                       instrument => store.dispatch(updateMarketfeedInstrument(instrument)));
    }
  }, getRandomInt(200, 1000));
}

function PlaceSampleOrders(store, maxOrderCount) {
  const interval = setInterval(() => {
    const buySellId = getRandomInt(0,1) === 0 ? "buy":"sell";
    const ticketBuySell = document.getElementById(buySellId);
    const orderCount = store.getState().orders.length;
    if(ticketBuySell != null && orderCount < maxOrderCount) {
      const confirmFunc = window.confirm;
      window.confirm = () => true;
      ticketBuySell.click();
      window.confirm = confirmFunc;
    } else {
      clearInterval(interval);
    }
  }, getRandomInt(300, 500));
}

/**
 * [Apps Application Main defining routes, separating between header and content]
 */
function App() {
  return (<HashRouter>
    <div>
      <ul className="header">
        <li>
          <NavLink exact={true} to="/">Trading Demo</NavLink>
        </li>
        <li>
          <NavLink to="/documentation">Documentation</NavLink>
        </li>
      </ul>
      <div className="content">
        <Route exact={true} path="/" component={TradingDemo}/>
        <Route path="/Documentation" component={Documentation}/>
      </div>
    </div>
  </HashRouter>)
}
