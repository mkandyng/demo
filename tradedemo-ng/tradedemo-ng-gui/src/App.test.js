import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import App from "./App";

/**
 * This is unit test of App
 * It leverage snapshot to test the presentation
 */

describe("App", () => {

    it("should render component comparing with previous snapshot shallow markup", () => {
        // Given

        // When
        const component = shallow(<App />);

        // Then
        expect(toJson(component)).toMatchSnapshot();
    });

});
