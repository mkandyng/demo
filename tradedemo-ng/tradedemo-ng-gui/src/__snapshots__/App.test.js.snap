// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render component comparing with previous snapshot shallow markup 1`] = `
<HashRouter>
  <div>
    <ul
      className="header"
    >
      <li>
        <NavLink
          exact={true}
          to="/"
        >
          Trading Demo
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/documentation"
        >
          Documentation
        </NavLink>
      </li>
    </ul>
    <div
      className="content"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/Documentation"
      />
    </div>
  </div>
</HashRouter>
`;
